pkg/envutil/envutil.go

```diff
@@ -78,8 +78,15 @@ func matchesPattern(name, pattern string) bool {
 		return true
 	}
 
-	prefix, found := strings.CutSuffix(pattern, "*")
-	return found && strings.HasPrefix(name, prefix)
+	if prefix, found := strings.CutSuffix(pattern, "*"); found {
```

Comment:
I've been wondering if once we support both prefix and postfix `*` if users would expect that infix `*` should also work?

We could simply replace every `*` with `.*`, and then anchor the pattern with `^pattern$` and perform a regex search. Probably should first make sure there are no other metadata characters in the pattern.

Anyways, not saying you should change this, I just want a discussion if we should support infix `*` or not? E.g. `PERL*LIB` would match `PERL5LIB` and `PERLLIB`.

Reply:
I actually saw your comment on the parent issue and thought I should make provision for it in this PR but I wanted to see what others would say before going ahead. Your suggestion makes perfect sense.

Reply:
Should be a separate commit, or perhaps a PR